src/hooks/useInputFocusing.ts

```diff
@@ -0,0 +1,36 @@
+import { useEffect } from 'react';
+
+interface UseInputFocusingProps {
+  refs: React.RefObject<HTMLInputElement>[];
+  deps: React.DependencyList;
+  maxLength: number;
+}
+
+const useInputFocusing = ({ refs, deps, maxLength }: UseInputFocusingProps) => {
+  useEffect(() => {
+    const focusedInput = refs.find((ref) => document.activeElement === ref.current);
+
```

Comment:
`(ref) => document.activeElement === ref.current` 콜백이 중복되어 발생하는게 조금 걸리네요. 유틸 함수로 분리해볼까요? findActiveElement 등으로 말이죠!

```suggestion
const findActiveElement = (ref) => document.activeElement === ref.current;

const focusedInput = refs.find(findActiveElement);
const focusedInputIndex = refs.findIndex(findActiveElement);
```

아니라면... findIndex 함수만 사용하여 `refs[index]` 접근하는 것도 방법입니다.

```suggestion
const focusedInputIndex = refs.findIndex((ref) => document.activeElement === ref.current);

if (focusedInputIndex === -1) return;

const focusedInput = refs[focusedInputIndex];
```

Reply:
둘 중 하나의 방법으로 선택해서 수정해보겠습니다!